charts/elasticsearch/templates/es-configmap.yaml

```diff
@@ -38,10 +38,19 @@ data:
         enabled: ${HTTP_CORS_ENABLE}
         allow-origin: ${HTTP_CORS_ALLOW_ORIGIN}
 
+    # Setting the ping schedule explictly ensures that
+    # tcp_keep_alive configuration does not cause components
+    # to have their connections terminated.
+    transport:
+      ping_schedule: "5s"
+
     discovery:
-      zen:
-        ping.unicast.hosts: ${DISCOVERY_SERVICE}
-        minimum_master_nodes: ${NUMBER_OF_MASTERS:2}
+      seed_hosts:
+        {{- $releaseName := $.Release.Name }}
+        {{- $replicas := .Values.master.replicas }}
+        {{- range $i, $e := until (int $replicas) }}
+        - {{ $releaseName }}-elasticsearch-master-{{ $i }}
+        {{- end }}
```

Comment:
I noticed deprecation notices in the logs for ping.unicast.hosts and minimum_master_nodes.